asf_levies_model/levies.py

```diff
@@ -2026,3 +2030,552 @@ def calculate_gbis_rate(
         else:
             raise ValueError("Insufficient information to calculate GBIS rate.")
         return rate
+
+
+class LevyCollection:
+    """A container for Levy objects.
+
+    Convenient abstraction for working with multiple levies.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        short_name: str,
+        levies: list,
+        denominators: Union[Dict[str, int], Dict[str, Dict[str, int]]],
+    ) -> None:
+        """Initializes the Levy collection instance based on list of provided levies.
+
+        Args:
+            name: Name for LevyCollection instance.
+            short_name: Abbreviation for LevyCollection instance.
+            levies: List of Levy objects.
+            denominators: A single set of denominators to apply to all levies, or a dictionary of levy specific denominators.
+        """
+        self.name = name
+        self.short_name = short_name
+        self.levies = levies
+        self.denominators = denominators
+
+    @property
+    def price_cap_period(self) -> PriceCapPeriod:
+        return self.levies[0].price_cap_period
```

Comment:
```suggestion
        if not all(
            levy.price_cap_period == self.levies[0].price_cap_period
            for levy in self.levies
        ):
            raise ValueError("All levies must have the same price_cap_period")
        return self.levies[0].price_cap_period
```

Open to discussion about this as I'm not sure if we will always want the levies to be of the same price cap period? However, if this is true, I think it's worth safeguarding against inconsistent price cap periods when instantiating the individual levies.

Reply:
To be honest, the only situation I can see where you might reasonably have different price cap periods would be in a time-series analysis context. In that setting I'd prefer to create a specific Collection object that handled that.

For me, I think this basic collection should apply solely to a specific price cap period.

Reply:
This is a good suggestion - commit it!

Reply:
Just realised that the check and error handle need to happen inside __init__ for the error to raised at point of instantiation. Will insert into the right place and commit then.